src/ResourceManagement/ContainerService/Domain/KubernetesCluster/Definition/IDefinition.cs

```diff
@@ -357,6 +359,21 @@ public interface IWithAddOnProfilesBeta  :
         /// </summary>
         /// <param name="addOnProfileMap">The cluster's add-on's profiles.</param>
         /// <return>The next stage of the update.</return>
-        Microsoft.Azure.Management.ContainerService.Fluent.KubernetesCluster.Update.IUpdate WithAddOnProfiles(IDictionary<string,Microsoft.Azure.Management.ContainerService.Fluent.Models.ManagedClusterAddonProfile> addOnProfileMap);
+        Microsoft.Azure.Management.ContainerService.Fluent.KubernetesCluster.Update.IUpdate WithAddOnProfiles(IDictionary<string, Microsoft.Azure.Management.ContainerService.Fluent.Models.ManagedClusterAddonProfile> addOnProfileMap);
+    }
+
+
+    /// <summary>
+    /// The stage of the Kubernetes cluster definition allowing to enable virtual node.
+    /// </summary>
+    public interface IWithVirtualNode :
+        Microsoft.Azure.Management.ResourceManager.Fluent.Core.IBeta
+    {
+        /// <summary>
+        /// Create a virtual node with ACI.
+        /// </summary>
+        /// <param name="subnetName">The subnet in AKS network to create virtual node.</param>
+        /// <returns>The next stage of the definition.</returns>
+        Microsoft.Azure.Management.ContainerService.Fluent.KubernetesCluster.Definition.IWithCreate WithVirtualNode(string subnetName);
```

Comment:
It might actually conflix with `IWithAddOnProfiles` stage, which set the whole `Inner.AddonProfiles`. The least to do is docstring mentioning this should be called before `WithAddOnProfiles`.